### PR TITLE
Audio block capability now enables/disables media upload's media sources

### LINKED ISSUE #3434: [Audio block] Disable media upload options on free sites
**Is your feature request related to a problem? Please describe.**

This feature is necessary since, on dot com, free plans don't have access to audio uploads/hosting. 

**Describe the solution you'd like**

Since the behavior isn't available, the functionality, media upload, and WP Media Library that provides them should be disabled/removed. 

**Additional context**
The Audio block will still be able to function with the URL option.

### PR DESCRIPTION
**Pull request by jd-alexander**
Fixes #3434

`WordPress iOS` https://github.com/wordpress-mobile/WordPress-iOS/pull/16533
`WordPress Android` https://github.com/wordpress-mobile/WordPress-Android/pull/14680
`Gutenberg` https://github.com/WordPress/gutenberg/pull/31966

To test:
https://github.com/WordPress/gutenberg/pull/31966

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

**Comment by jd-alexander** [comment]
@fluiddot I am going to be finalizing these set of PRs soon with the capabilities changes. While working on this on Friday, I noticed the translation related failure while trying to build the bundle to test on iOS. I am assuming that I have to utilize the `string-extract-fix` patch and localization files from 1.52.2 you created for the previous release to get this working as expected. Let me know what you think.

**Comment by fluiddot** [comment]
> @fluiddot I am going to be finalizing these set of PRs soon with the capabilities changes. While working on this on Friday, I noticed the translation related failure while trying to build the bundle to test on iOS. I am assuming that I have to utilize the `string-extract-fix` patch and localization files from 1.52.2 you created for the previous release to get this working as expected. Let me know what you think.

Yeah, I'd recommend you to use the patch and the localization files from `1.52.2` to produce the proper localization strings files. However, if you're not introducing new strings, probably you could skip modifying those files in this PR and use the ones we currently have on `develop` branch.

Btw, I've just checked the [commit "Merge branch 'develop' into add/audio-block-capability"](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3523/commits/91024f951accb44159e6023875b5dd1b307509bf) and I saw that some of the strings have been pushed with conflicts.

**Comment by jd-alexander** [comment]
> Yeah, I'd recommend you to use the patch and the localization files from 1.52.2 to produce the proper localization strings files. However, if you're not introducing new strings, probably you could skip modifying those files in this PR and use the ones we currently have on develop branch.

Thanks for the tips. I haven't updated this PR as yet as I was investigating the hermes crash which is a blocker for this PR since I would need to sync with the latest `gb-mobile` `develop` and gb `trunk`.

**Comment by jd-alexander** [comment]
Since there are changes within the main apps PR apart from the `gutenberg-mobile` ref changes, I will be creating  a `gutenberg-mobile` tag of `v1.54.0-alpha2` when this PR is approved.

@fluiddot when I run `npm run bundle` with or without the patch I was getting `genstring` failures so I am not sure if the translation changes here aren't to be expected. Let me know. Thanks! I am going to give the PR you created and related discussions a thorough review to understand how translations work because I haven't had a look at the code generation logic in detail.

**Comment by jd-alexander** [comment]
I did re-runs of `Gutenberg Editor Slash Inserter tests` but I am still getting failures. The changes here are unrelated as far as I am concerned.

**Comment by fluiddot** [comment]
> Since there are changes within the main apps PR apart from the `gutenberg-mobile` ref changes, I will be creating a `gutenberg-mobile` tag of `v1.54.0-alpha2` when this PR is approved.

👍 

> @fluiddot when I run `npm run bundle` with or without the patch I was getting `genstring` failures so I am not sure if the translation changes here aren't to be expected. Let me know. Thanks! I am going to give the PR you created and related discussions a thorough review to understand how translations work because I haven't had a look at the code generation logic in detail.

Yikes, could you share the failures that you got when generating the bundle?

I see that these changes have a new string (`Insert from URL`) in the code but I verified that it's already in the Gutenberg translations ([GlotPress reference](https://translate.wordpress.org/projects/wp-plugins/gutenberg/dev/en-gb/default/?filters%5Bterm%5D=Insert+from+URL&filters%5Bterm_scope%5D=scope_any&filters%5Bstatus%5D=current_or_waiting_or_fuzzy_or_untranslated&filters%5Buser_login%5D=&filter=Apply+Filters&sort%5Bby%5D=priority&sort%5Bhow%5D=desc)) so it's expected that no modification is made to the localization strings files that we push to the main apps.

- `bundle/android/strings.xml`
- `bundle/ios/GutenbergNativeTranslations.swift`

However, I saw the translation files (`i18n-data` files) have been modified and they are missing strings (you can check that by opening any of them ([es.json](https://raw.githubusercontent.com/wordpress-mobile/gutenberg-mobile/3a7ec29196c4afadc6f5897b1ed2fe131b6f155f/bundle/ios/assets/gutenberg/packages/react-native-editor/i18n-cache/data/es.json)) and look for the word `block title`). I'll apply the fixes from the [PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3555) that should help us with this issue.

**Comment by fluiddot** [comment]
> I did re-runs of `Gutenberg Editor Slash Inserter tests` but I am still getting failures. The changes here are unrelated as far as I am concerned.

This was fixed yesterday via this [PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3549) so if you update the branch with `develop` it should be fixed 🎊 .

**Comment by fluiddot** [comment]
@jd-alexander Heads up that I've updated all the PRs for doing final testing, the only changes I made are:
- [Fix indent issues on Gutenberg props](https://github.com/WordPress/gutenberg/pull/31966/commits/11a79e7cd70695b06ddaf0dcf81a63defd2066b7)
- Merge conflicts with `develop`

Let me know if you have any concerns regarding these updates, thanks 🙇 !

**Comment by jd-alexander** [comment]
> Let me know if you have any concerns regarding these updates, thanks 🙇 !

I have no issue with these updates. Thanks @fluiddot 🙇🏾

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -1,3 +1,7 @@
+Unreleased
+------
+* [***] Audio block now available on WP.com sites on the free plan. [https://github.com/wordpress-mobile/gutenberg-mobile/pull/3523]
```
Comment (fluiddot) [inline RELEASE-NOTES.txt:3]: I moved this entry to the unreleased section as this PR won't be included in `1.54.0`.

**Comment by fluiddot** [comment]
I changed the milestone of this PR to `1.55.0 (17.6)` as the `1.54.0` version is already cut.

**Review by fluiddot — APPROVED** [review]
Approved via https://github.com/WordPress/gutenberg/pull/31966#pullrequestreview-673060329.

**Pull request merged**